etl/data_helpers/geo.py

```diff
@@ -592,12 +605,21 @@ def list_members_of_region(
     )
 
     if ds_income_groups is not None:
-        # Get the main table from the income groups dataset.
-        df_income = pd.DataFrame(ds_income_groups["wb_income_group"]).reset_index()
+        if "wb_income_group" in ds_income_groups.table_names:
+            # TODO: Remove this block once the old income groups dataset has been archived.
+            # Get the main table from the income groups dataset.
+            df_income = (
+                pd.DataFrame(ds_income_groups["wb_income_group"])
+                .reset_index()
+                .rename(columns={"income_group": "classification"})
+            )
+        else:
+            # Get the table with the current definitions of income groups.
+            df_income = ds_income_groups["income_groups_latest"].reset_index()
```

Comment:
should probably check that `income_groups_latest in ds_income_groups.table_names` and then have an `else` clause raising an error saying something like "table was not found"